notify/notify.go

```diff
@@ -375,6 +375,7 @@ func (pb *PipelineBuilder) New(
 	inhibitor *inhibit.Inhibitor,
 	silencer *silence.Silencer,
 	times map[string][]timeinterval.TimeInterval,
+	marker types.Marker,
```

Comment:
I initially tried to suggest the changes I thought were necessary but instead decided it'll be simpler to just do it - I've created the abstraction here: https://github.com/prometheus/alertmanager/pull/3556

This leaves the functionality intact so that you can then rebase on top of that to add the marker usage directly on the `intervener` (for the lack of a better name).